meteor/client/lib/multiSelect.tsx

```diff
@@ -58,74 +53,53 @@ export class MultiSelect extends React.Component<IProps, IState> {
 	}
 
 	refreshChecked() {
-		if (this.props.value && _.isArray(this.props.value)) {
-			const checkedValues: _.Dictionary<boolean> = {}
-			_.forEach(this.props.value, (value) => {
-				checkedValues[value] = true
-			})
+		this.setState({
+			checkedIds: this.props.value.map((option) => option.value),
+		})
+	}
 
-			this.setState({
-				checkedValues,
-			})
+	handleChange = (clickedOption: MultiSelectOption) => {
+		const checkedIds = this.state.checkedIds
+
+		const index: number = checkedIds.findIndex((id) => id === clickedOption.value)
+		if (index === -1) {
+			checkedIds.push(clickedOption.value)
 		} else {
-			this.setState({
-				checkedValues: {},
-			})
+			checkedIds.splice(index, 1)
 		}
-	}
-
-	handleChange = (item) => {
-		const obj = {}
-		obj[item] = !this.state.checkedValues[item]
-		const valueUpdate = _.extend(this.state.checkedValues, obj)
 
 		this.setState({
-			checkedValues: valueUpdate,
+			checkedIds: checkedIds,
 		})
 
-		if (this.props.onChange && typeof this.props.onChange === 'function') {
-			this.props.onChange({
-				selectedValues: _.compact(
-					_.values(
-						_.mapObject(valueUpdate, (value, key) => {
-							return value ? key : null
-						})
-					)
-				),
-			})
+		if (!this.props.onChange || typeof this.props.onChange !== 'function') {
+			return
 		}
+		this.props.onChange(this.getOptionsFromCheckedIds())
+	}
+
+	getOptionsFromCheckedIds(): MultiSelectOption[] {
+		return this.state.checkedIds.map((id) => this.props.options.find((option) => option.value === id)!)
```

Comment:
> because of this type error you are ignoring?

I assume you mean the `!` at the end?

If an option is deleted then `EditAttributeMultiSelect` compares what is saved in the database against the options provided. If the options provided does not contain the saved value, then a "missing" option is being added to options array and then parsed to `MultiSelect`.

Snippet from `EditAttributeMultiSelect`:
```
const availableOptions = this.getAvailableOptions()
const missingOptions = this.getMissingOptions(availableOptions)
...
<MultiSelect
  ...
  options={availableOptions.concat(missingOptions)}
```

I agree it's not good that `MultiSelect` is reliant on `EditAttributeMultiSelect` to avoid a undefined error. I will see what I can do about that.